src/um/wincodec.rs

```diff
@@ -1,373 +1,1973 @@
+        dither: WICBitmapDitherType,
+        pIPalette: *const IWICPalette,
+        alphaThresholdPercent: c_double,
         paletteTranslate: WICBitmapPaletteType
     ) -> HRESULT,
     fn CanConvert(
-        srcPixelFormat: REFWICPixelFormatGUID, dstPixelFormat: REFWICPixelFormatGUID,
+        srcPixelFormat: REFWICPixelFormatGUID,
+        dstPixelFormat: REFWICPixelFormatGUID,
         pfCanConvert: *mut BOOL
     ) -> HRESULT
 });
-RIDL!(#[uuid(0xbebee9cb, 0x83b0, 0x4dcc, 0x81, 0x32, 0xb0, 0xaa, 0xa5, 0x5e, 0xac, 0x96)]
-interface IWICPlanarFormatConverter(IWICPlanarFormatConverterVtbl)
-    : IWICBitmapSource(IWICBitmapSourceVtbl) {
+
+RIDL!(#[uuid(0xBEBEE9CB, 0x83B0, 0x4DCC, 0x81, 0x32, 0xB0, 0xAA, 0xA5, 0x5E, 0xAC, 0x96)]
+interface IWICPlanarFormatConverter(IWICPlanarFormatConverterVtbl): IWICBitmapSource(IWICBitmapSourceVtbl) {
     fn Initialize(
-        ppPlanes: *mut *mut IWICBitmapSource, cPlanes: UINT,
-        dstFormat: REFWICPixelFormatGUID, dither: WICBitmapDitherType, pIPalette: *mut IWICPalette,
-        alphaThreasholdPercent: f64, paletteTranslate: WICBitmapPaletteType
+        ppPlanes: *const *const IWICBitmapSource,
+        cPlanes: UINT,
+        dstFormat: REFWICPixelFormatGUID,
+        dither: WICBitmapDitherType,
+        pIPalette: *const IWICPalette,
+        alphaThresholdPercent: c_double,
+        paletteTranslate: WICBitmapPaletteType
     ) -> HRESULT,
     fn CanConvert(
-        pSrcPixelFormats: *const WICPixelFormatGUID, cSrcPlanes: UINT,
-        dstPixelFormat: REFWICPixelFormatGUID, pfCanConvert: *mut BOOL
+        pSrcPixelFormats: *const WICPixelFormatGUID,
+        cSrcPlanes: UINT,
+        dstPixelFormat: REFWICPixelFormatGUID,
+        pfCanConvert: *mut BOOL
     ) -> HRESULT
 });
+
 RIDL!(#[uuid(0x00000302, 0xa8f2, 0x4877, 0xba, 0x0a, 0xfd, 0x2b, 0x66, 0x45, 0xfb, 0x94)]
 interface IWICBitmapScaler(IWICBitmapScalerVtbl): IWICBitmapSource(IWICBitmapSourceVtbl) {
     fn Initialize(
-        pISource: *mut IWICBitmapSource, uiWidth: UINT, uiHeight: UINT,
+        pISource: *const IWICBitmapSource,
+        uiWidth: UINT,
+        uiHeight: UINT,
         mode: WICBitmapInterpolationMode
     ) -> HRESULT
 });
-RIDL!(#[uuid(0xe4fbcf03, 0x223d, 0x4e81, 0x93, 0x33, 0xd6, 0x35, 0x55, 0x6d, 0xd1, 0xb5)]
+
+RIDL!(#[uuid(0xE4FBCF03, 0x223D, 0x4e81, 0x93, 0x33, 0xD6, 0x35, 0x55, 0x6D, 0xD1, 0xB5)]
 interface IWICBitmapClipper(IWICBitmapClipperVtbl): IWICBitmapSource(IWICBitmapSourceVtbl) {
-    fn Initialize(pISource: *mut IWICBitmapSource, prc: *const WICRect) -> HRESULT
+    fn Initialize(
+        pISource: *const IWICBitmapSource,
+        prc: *const WICRect
+    ) -> HRESULT
 });
-RIDL!(#[uuid(0x5009834f, 0x2d6a, 0x41ce, 0x9e, 0x1b, 0x17, 0xc5, 0xaf, 0xf7, 0xa7, 0x82)]
-interface IWICBitmapFlipRotator(IWICBitmapFlipRotatorVtbl)
-    : IWICBitmapSource(IWICBitmapSourceVtbl) {
+
+RIDL!(#[uuid(0x5009834F, 0x2D6A, 0x41ce, 0x9E, 0x1B, 0x17, 0xC5, 0xAF, 0xF7, 0xA7, 0x82)]
```

Comment:
Horrible mistake manual numeric editing. Want to follow the original description.